### PR TITLE
A draft machine

### PR DESCRIPTION
**Pull request by markrwilliams**
@moshez: it's remarkable for what it *isn't*.  Hope this helps you!

### CODE REVIEW

File: txreviewbot/_machine.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+@attr.s
+class RemoveLabel(object):
+    name = attr.ib()
+
+
+class ReviewMachine(object):
+    _machine = MethodicalMachine()
+
+    @_machine.input()
+    def review(self, pullRequest):
```
Comment (moshez) [inline txreviewbot/_machine.py:19]: Do we really need the "pullRequest" object here?
(Ditto for other inputs)

File: txreviewbot/_machine.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+        """
+
+    @_machine.state()
+    def merged(self):
+        """
+
+        """
+
+    @_machine.output()
```
Comment (moshez) [inline txreviewbot/_machine.py:90]: An alternative is to have no outputs on most transitions, and just have a 

    @_machine.input()
    def getRemediationPlan(self, pullRequest):

And have a per-state output that returns the remediation plan.

**Review by moshez — APPROVED** [review]
I think you should merge this anyway. This has a lot of important stuff, and nobody is depending on our repo. We can start opening issues for things we want to address, I guess? I think our first milestone should be "self-hosting" :)

File: txreviewbot/_machine.py (hunk 1 above)
Comment (markrwilliams) [reply txreviewbot/_machine.py:19]: In the real version, no!  But for now, I'm not sure what we want out of the pull request for each input, so I'd like to keep this as a placeholder.

File: txreviewbot/_machine.py (hunk 2 above)
Comment (markrwilliams) [reply txreviewbot/_machine.py:90]: Ah, of course!  I'm going to land this in favor of a follow-up PR that uses this approach.

**Pull request merged**